### PR TITLE
thunderbird: Enable official branding by default

### PR DESCRIPTION
**Pull request by LouisDK1**
###### Motivation for this change

Using official branding for Mozilla applications with minor modifications is now allowed as mentioned in https://github.com/NixOS/nixpkgs/issues/31843 by @sylvestre - a Mozilla/Debian developer

Because of this Firefox now use official branding in NixOS and I suggest we do the same thing with Thunderbird.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by LouisDK1** [comment]
@rycee If you agree or have anything to add feel free.

**Comment by prusnak** [comment]
Maybe you should update the comment above the modified line to reflect upon the current situation and include a link to https://github.com/NixOS/nixpkgs/issues/31843#issuecomment-346372756

**Review by lovesegfault — APPROVED** [review]
Add a reference as @prusnak asked, but other than that :shipit:

**Comment by LouisDK1** [comment]
@prusnak @lovesegfault Done

**Review by lovesegfault — APPROVED** [review]
Let's land this!

**Comment by LouisDK1** [comment]
I made a white space typo. Should be fixed now.

**Review by prusnak — APPROVED** [review]

**Comment by pbogdan** [comment]
Have we received an explicit permission to use official branding for Thunderbird specifically?

**Comment by LouisDK1** [comment]
@pbogdan
> Have we received an explicit permission to use official branding for Thunderbird specifically?

In this issue: https://github.com/NixOS/nixpkgs/issues/31843 @sylvestre approve the nix-related changes for Firefox and states that the situation is similar to that of Debian. In the same post he shares this link: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=815006 where he have made this statement: "= About Icedove/Thunderbird =
Similar discussions with the icedove maintainers have just started to do the same
for this package."

If that's not enough I could reach out to him.

**Comment by LouisDK1** [comment]
I don't know why this fails:
undefined variable 'enableOfficialBranding' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/packet-spot-eval-3/pkgs/applications/networking/mailreaders/thunderbird/68.nix:74:1

I'm defining the variable at line 74 so I don't know why I get that error.

**Comment by sylvestre** [comment]
@LouisDK1 please keep in mind that it also depends on the patches applied on top of Fx or Tb

**Comment by LouisDK1** [comment]
@sylvestre 
> @LouisDK1 please keep in mind that it also depends on the patches applied on top of Fx or Tb

I am aware of that.
If you want you're welcome to take a look at the build script and see if it could be approved in relation to the branding.

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix
Comment (prusnak) [inline pkgs/applications/networking/mailreaders/thunderbird/default.nix:63]:
```suggestion
# As stated by Sylvestre Ledru (@sylvestre) on Nov 22, 2017 at
```

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix
Comment (prusnak) [inline pkgs/applications/networking/mailreaders/thunderbird/default.nix:72]:
```suggestion
, enableOfficialBranding ? true
```

File: pkgs/applications/networking/mailreaders/thunderbird/68.nix
Comment (prusnak) [inline pkgs/applications/networking/mailreaders/thunderbird/68.nix:74]:
```suggestion
, enableOfficialBranding ? true
```

File: pkgs/applications/networking/mailreaders/thunderbird/68.nix
Comment (prusnak) [inline pkgs/applications/networking/mailreaders/thunderbird/68.nix:65]:
```suggestion
# As stated by Sylvestre Ledru (@sylvestre) on Nov 22, 2017 at
```

**Review by prusnak — CHANGES_REQUESTED** [review]
Let's use this opportunity to fix very weird use of commas too.

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix
Comment (prusnak) [reply pkgs/applications/networking/mailreaders/thunderbird/default.nix:63]: These are not addressed in your PR. Did you forgot to force push?

Comment (LouisDK1) [reply pkgs/applications/networking/mailreaders/thunderbird/default.nix]: I've force pushed now.

**Review by prusnak — APPROVED** [review]
LGTM

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix
Comment (flokli) [inline pkgs/applications/networking/mailreaders/thunderbird/default.nix:79]: can this be two separate commits, one for the bump, one for the branding? Can be in the same PR, otherwise LGTM.

Comment (flokli) [reply pkgs/applications/networking/mailreaders/thunderbird/default.nix:79]: Okay, turns out, these versions are already like this on master. I'll rebase on latest master.

**Review by flokli — APPROVED** [review]

**Comment by vcunat** [comment]
@sylvestre wrote:

> could you please tell me who reviewed the nix patches on top of tb?

If there's been no explicit permission (and I can't see any hint that there has), I personally find it surprising that this got merged.  I'm no good at law stuff, though.

**Comment by sylvestre** [comment]
As a Mozilla representative, I am indeed surprised that it got merged without a confirmation that the patches are safe.
Please revert until you have a proper sign off.

**Comment by vcunat** [comment]
OK, reverted.

So... let me ask [again](https://github.com/NixOS/nixpkgs/pull/55026#issuecomment-564975732): how can we progress towards getting a permission?

**Comment by sylvestre** [comment]
Please provide the url to the list of patches which will applied to tb with some docs if they aren't self explanatory.

**Comment by vcunat** [comment]
@sylvestre, I'm describing current status; I'm not aware of any changes planned around that.

[The only patch](https://github.com/NixOS/nixpkgs/blob/c8e727c78339db22941ae53c306dbefd2fc5335b/pkgs/applications/networking/mailreaders/thunderbird/no-buildconfig.patch) seems quite simple to me.  Its purpose is that the page would cause unnecessary runtime dependencies (e.g. on clang) due to the way they get discovered automatically by Nix.

Configure flags and the rest of "build recipe", in case they mattered: https://github.com/NixOS/nixpkgs/blob/c8e727c78339db22941ae53c306dbefd2fc5335b/pkgs/applications/networking/mailreaders/thunderbird/default.nix#L207-L247

**Comment by flokli** [comment]
I wasn't aware still was still somewhat unclear. Thanks @sylvestre  for getting back to as, and thanks @vcunat for doing the revert for now.

Regarding that patch - these seem to be pretty similar to the no-buildconfig patches [we already have for firefox](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/networking/browsers/firefox) - and given we build firefox with official branding, I'd assume those have been approved by Mozilla of some sort.

Hopefully this speeds up the review ;-)

**Comment by sylvestre** [comment]
Yeah, looks good to me. ship it!
Just one thing, please add a comment in the patch explaining the why buildconfig is disabled.

**Comment by LouisDK1** [comment]
If someone is able to give a statement on why build config is disabled I'll update the comment and update this PR.

**Comment by vcunat** [comment]
It seemed easier to do it directly.  Patch explained in c09965182 and branding enabled in f77e427cf.

**Pull request merged**
